test/application/respond.js

```diff
@@ -234,6 +234,7 @@ describe('app.respond', () => {
       app.use((ctx, next) => {
         const res = ctx.res;
         ctx.status = 200;
+        ctx.respond = false;
```

Comment:
It's not necessary here because there is no more middleware this application used. Also it is testing the app.respond method to handle this situation, if `ctx.respond` was set to false, we can't test this situation.

Reply:
Unless this is set to false the test will call ‘res.end()’ twice and with data the second time. That will cause a write after end error in future versions of node...

Reply:
See, https://github.com/nodejs/node/pull/28666

Reply:
> Unless this is set to false the test will call ‘res.end()’ twice and with data the second time. That will cause a write after end error in future versions of node...

Then this test case will be meaningless. I think you should remove this line `setTimeout(() => res.end('Goodbye'), 0);` to fix this.

Reply:
sorted